File: platform/src/java/com/xcase/rest/generator/raml/IScanner.java
package com.xcase.rest.generator.raml;

import com.google.gson.*;
import com.xcase.common.impl.simple.core.CommonHTTPManager;
import com.xcase.common.utils.ConverterUtils;
import com.xcase.rest.generator.*;
import com.xcase.rest.generator.raml.EventType;
import com.xcase.rest.generator.raml.IParsingEventVisitor;
import com.xcase.rest.generator.raml.Mark;
import com.xcase.rest.generator.raml.tokens.Token;
import java.lang.invoke.MethodHandles;
import java.io.*;
import java.net.*;
import java.nio.file.*;
import java.util.*;
import org.apache.http.HttpEntity;
import org.apache.http.HttpResponse;
import org.apache.http.NameValuePair;
import org.apache.http.message.BasicNameValuePair;
import org.apache.http.util.EntityUtils;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public interface IScanner
{
    Token getCurrent();
    /// <summary>
    /// Moves to the next token and consumes the current token.
    /// </summary>
    boolean moveNext();

    /// <summary>
    /// Moves to the next token without consuming the current token.
    /// </summary>
    boolean moveNextWithoutConsuming();

    /// <summary>
    /// Consumes the current token.
    /// </summary>
    void consumeCurrent();
}

File: XCase.Swagger.ProxyGenerator/RAML/IScanner.cs
namespace XCase.REST.ProxyGenerator.RAML
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;
    using System.Threading.Tasks;
    using XCase.REST.ProxyGenerator.RAML.Tokens;

    public interface IScanner
    {
        /// <summary>
        /// Gets the current position inside the input stream.
        /// </summary>
        /// <value>The current position.</value>
        Mark CurrentPosition
        {
            get;
        }

        /// <summary>
        /// Gets the current token.
        /// </summary>
        Token Current
        {
            get;
        }

        /// <summary>
        /// Moves to the next token and consumes the current token.
        /// </summary>
        bool MoveNext();

        /// <summary>
        /// Moves to the next token without consuming the current token.
        /// </summary>
        bool MoveNextWithoutConsuming();

        /// <summary>
        /// Consumes the current token.
        /// </summary>
        void ConsumeCurrent();
    }
}
